doc/api/single-executable-applications.md

```diff
@@ -0,0 +1,88 @@
+# Single executable applications
+
+<!--introduced_in=REPLACEME-->
+
+> Stability: 1 - Experimental: This feature is currently being designed and will
+> still change.
+
+<!-- source_link=lib/internal/main/single_executable_application.js -->
+
+Node.js supports the creation of [single executable applications][] by allowing
+the injection of a JavaScript file into the `node` binary. During start up, the
+program checks if a resource (on [PE][]) or section (on [Mach-O][]) or note (on
+[ELF][]) named `NODE_JS_CODE` exists (on macOS, in the `NODE_JS` segment). If it
+is found, it executes its contents, otherwise it operates like plain Node.js.
```

Comment:
I appreciate the detail here, but perhaps worth shifting this level of detail further down rather than in the intro of the section. Alternatively, moving the next paragraph above this one could achieve the same goal with less work.

Reply:
@bnb done, PTAL!